.github/workflows/base_tests.yml

```diff
@@ -0,0 +1,130 @@
+name: Base Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_call:
+    inputs:
+      include:
+        description: 'Include matrix configurations'
+        required: false
+        type: string
+        default: "[]"
+      exclude:
+        description: 'Exclude matrix configurations'
+        required: false
+        type: string
+        default: "[]"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  delete_cache:
+    name: Delete github action's branch cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: snnaplab/delete-branch-cache-action@v1
+
+  test:
+    needs: delete_cache
+    timeout-minutes: 90
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - os: linux
+            builder: ubuntu-22.04
+            cpu: amd64
+            shell: bash
+          - os: linux
+            builder: ubuntu-22.04
+            cpu: i386
+            shell: bash
+          - os: macos
+            builder: macos-13
+            cpu: amd64
+            shell: bash
+          - os: windows
+            builder: windows-2022
+            cpu: amd64
+            shell: msys2 {0}
+        nim_branch:
+          - version: version-1-6
+            memory_management: refc
+          - version: version-2-0
+            memory_management: refc
+          # - version: devel
```

Comment:
Could you please explain why this comment is necessary?

Reply:
The `devel` version is not used in the default matrix (`base_tests.yml`), but it's used in one of the daily runs. To accomplish that we use the `include` parameter, and because that's used in a different file I thought it'd be useful keeping it as comment in the original matrix. That way it serves as documentation of what's in use and a guideline for "nim environment" configurations for other workflows.

Would you prefer if I added a clarifying comment mentioning its purpose?

Reply:
I'd prefer if you could please remove all commented code.

Reply:
But comments explaing why include and exclude were added and how they are used could be helpful.

Reply:
I wouldn't mind removing this one, although I would argue it provides context (this combination is used somewhere else as stated above).
But I would definitely avoid adding a comment explaining an already-documented Github's CI's feature.